build/build-resources/reformat-modified

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+#
+# Reformats java files that have been added or modified, according to git, using Maven
+#
+# SPDX-License-Identifier: Apache-2.0 OR EPL-2.0
+# Copyright 2023 Christian Kohlschuetter, christian@kohlschutter.com
+#
+# Usage: reformat-modified [git-diff-args]
+#
+
+cd $(dirname $0)/../..
+
+# In a multi-module Maven project setup, the files are relative to each submodule,
+# so we have to wildcard the path relative from each submodule.
+includes=($(for f in $(git diff --name-only $@ | grep \.java); do
+  if [[ -z "${f##*/src/*}" ]]; then
+      echo **/${f##*/src/},
+  else
+      echo "$f,"
+  fi 
+done)
+$(git ls-files --others --exclude-standard | grep \.java )
```

Comment:
This should only modify changed lines and import section, no other lines in the source file should be altered.

(the Intellij formatter supports this)

Reply:
This is as much as I'm able to contribute at this point.

I'm really trying to help here, but I feel you set the bar too high, @joakime.
I'm not going to rewrite the Eclipse formatter for you.